src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/MirrorEndpointsSpec.groovy

```diff
@@ -0,0 +1,577 @@
+
+        then: "Mirror point is no longer present in flow and mirror APIs"
+        Wrappers.wait(WAIT_OFFSET) { assert northboundV2.getFlow(flow.flowId).mirrorPathStatuses.empty }
+        northboundV2.getAllFlows()[0].mirrorPathStatuses.empty
+        northboundV2.getMirrorPoints(flow.flowId).points.empty
+
+        and: "Mirror flow rule is no longer bind to OF group"
+        getFlowRule(swPair.src.dpId, flowPathDirection).instructions.applyActions.group == null
+
+        and: "OF group is removed"
+        !fl.getGroupStats(swPair.src.dpId).group.find { it.groupNumber == groupId }
+
+        and: "Src switch and flow pass validation"
+        northbound.validateSwitch(swPair.src.dpId).verifyRuleSectionsAreEmpty(swPair.src.dpId,
+                ["missing", "misconfigured"])
+        northbound.validateFlow(flow.flowId).each { direction -> assert direction.asExpected }
+
+        when: "Delete the flow"
+        flowHelperV2.deleteFlow(flow.flowId)
+
+        then: "Src switch pass validation"
+        northbound.validateSwitch(swPair.src.dpId).verifyRuleSectionsAreEmpty(swPair.src.dpId)
+        def testDone = true
+
+        cleanup:
+        !testDone && flow && flowHelperV2.deleteFlow(flow.flowId)
+
+        where:
+        mirrorDirection << [MirrorPointDirection.FORWARD, MirrorPointDirection.REVERSE]
+        //flowPathDirection required to pick correct ingress/egress rule that should contain OF group information
+        flowPathDirection << [FlowPathDirection.FORWARD, FlowPathDirection.REVERSE]
+    }
+
+    @Tidy
+    def "Can create a mirror point on the same port as flow, different vlan, #mirrorDirection"() {
+        given: "A flow"
+        def swPair = topologyHelper.switchPairs[0]
+        def flow = flowHelperV2.randomFlow(swPair)
+        flowHelperV2.addFlow(flow)
+
+        when: "Add a mirror point on the same port with flow, different vlan"
+        def mirrorPoint = FlowMirrorPointPayload.builder()
+                .mirrorPointId(flowHelperV2.generateFlowId())
+                .mirrorPointDirection(mirrorDirection.toString())
+                .mirrorPointSwitchId(swPair.dst.dpId)
+                .sinkEndpoint(FlowEndpointV2.builder().switchId(flow.destination.switchId)
+                        .portNumber(flow.destination.portNumber)
+                        .vlanId(flow.destination.vlanId - 1)
+                        .build())
+                .build()
+        northboundV2.createMirrorPoint(flow.flowId, mirrorPoint)
+
+        then: "Mirror point is successfully created"
+        Wrappers.wait(WAIT_OFFSET) {
+            northboundV2.getFlow(flow.flowId).mirrorPathStatuses[0].status == FlowPathStatus.ACTIVE.toString().toLowerCase()
+        }
+
+        when: "Delete the flow without deleting its mirror point"
+        flowHelperV2.deleteFlow(flow.flowId)
+
+        then: "Mirror point is also deleted from db"
+        database.getMirrorPoints().empty
```

Comment:
I can only get mirror points for certain existing flow. Since the flow is deleted at this point, I can't use that API. Thus making sure that the mirrorPoint is indeed removed in database directly